code/datums/actions/item_action.dm

```diff
@@ -33,27 +35,32 @@
 	return TRUE
 
 /datum/action/item_action/update_button_icon()
-	button.overlays.Cut()
-	var/obj/item/I = target
-	var/old_layer = I.layer
-	var/old_plane = I.plane
-	I.layer = ABOVE_HUD_LAYER
-	I.plane = ABOVE_HUD_PLANE
-	button.overlays += I
-	I.layer = old_layer
-	I.plane = old_plane
+	if(use_obj_appeareance)
+		if(visual_references[VREF_MUTABLE_LINKED_OBJ])
+			button.cut_overlay(visual_references[VREF_MUTABLE_LINKED_OBJ])
+		var/obj/item/I = target
+		var/item_image = mutable_appearance(I.icon, I.icon_state, ABOVE_HUD_LAYER, HUD_PLANE)
+		visual_references[VREF_MUTABLE_LINKED_OBJ] = item_image
+		button.add_overlay(item_image)
+	else if(visual_references[VREF_MUTABLE_LINKED_OBJ])
+		button.cut_overlay(visual_references[VREF_MUTABLE_LINKED_OBJ])
+		visual_references[VREF_MUTABLE_LINKED_OBJ] = null
```

Comment:
Dont see why you have the same code twice here for the cut instead of just moving it before the if